contract_invoice_merge_by_partner/models/account_analytic_analysis.py

```diff
@@ -12,19 +12,17 @@ class PurchaseOrderLine(models.Model):
     def _recurring_create_invoice(self, automatic=False):
         invoice_obj = self.env['account.invoice']
         invoices = invoice_obj.browse(
-            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic)
-        )
+            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic))
         res = []
         unlink_list = []
-        for partner in invoices.mapped('partner_id'):
+        for partner in set(invoices.mapped('partner_id')):
```

Comment:
You don't need a set, `mapped` already returns a recordset without duplicates. Even more, that can't be done, as the result in a recordset, not a list.